tests/test_merge.py

```diff
@@ -51,33 +52,214 @@ def test_merge_branch(wst, capsys):
     assert out.split('\n')[0] == 'Merging 3.0.x into master'
 
 
-def test_merge_branch_with_whitelist(wst, capsys):
+def run_commits_with_4(wst, capsys, to_skip, do_commit4):
```

Comment:
@sohinik this is an interesting approach to refactoring the git commits, and although it reduces the amount of code and repeated calls i think it has some draw backs. 
- it makes it a little harder to read the tests. when you read the tests now you really need to walk through this method each time to understand what is going to happen. for example looking at the array that gets passed in when i call this isn't clear what all the true/false means just from looking at it.
- you still have a bunch of code being repeated in with in this method as well.
- it doesn't provide you with much flexibility in the future if you want to do different scenarios, for example if we now want a test with 5 commits, or we want to change the branch we merge or merge several branches, or other scenarios.

I was thinking of something a little simpler where we just make a function for making the commits. A function like this one below. Maybe instead of skip being a boolean it could be the skip message in case we want to test on that, but maybe not needed.

```
def make_commit(name, skip=false):
  run('touch {}.txt'.format(name))
  run('git add -A')
  msg = name

  if skip:
    msg = "[skip] {}".format(msg)

  run('git commit -m {}'.format(msg))
  sha = run('git rev-parse --verify HEAD', return_output=True).strip()
  return sha
```